Build failure: netcdf-cxx4

### Steps To Reproduce

While rebuilding on this system that requires the package `nco` a failure while building `netcdf-cxx4` has been occurring since NixOS 22.11. One of the tests fails.

### Build log
```
-- Performing Test CXX_HAS_SHORTEN_64_32 - Failed                                                                                                                                                                    
# NetCDF C Configuration Summary                                                                                                                                                                                     
==============================                                                                                                                                                                                       
                                                                                                                                                                                                                     
# General                                                                                                                                                                                                            
-------                                                                                                                                                                                                              
NetCDF Version:         4.3.1                                                                                                                                                                                        
Configured On:          Sun Feb 26 15:18:14 UTC 2023                                                                                                                                                                 
Host System:            x86_64-Linux-5.15.63                                                                                                                                                                         
Build Directory:        /build/source/build                                                                                                                                                                          
Install Prefix:                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                                                                                                                                          
# Compiling Options                                                                                                                                                                                                                                                                                                                                                                                                                       
-----------------                                                                                                                                                                                                                                                                                                                                                                                                                         
C Compiler:             /nix/store/9yxrwp848f7msm6m2442yfpji8m3206b-gcc-wrapper-11.3.0/bin/gcc                                                                                                                                                                                                                                                                                                                                            
CFLAGS:                  -g -Wall -Wno-unused-variable -Wno-unused-parameter -O3 -DNDEBUG                                                                                                                            
CPPFLAGS:                -g -Wall -Wno-unused-variable -Wno-unused-parameter -O3 -DNDEBUG                                                                                                                            
LDFLAGS:                                                                                                                                                                                                             
AM_CFLAGS:                                                                                                                                                                                                           
AM_CPPFLAGS:                                                                                                                                                                                                         
AM_LDFLAGS:                                                                                                                                                                                                          
Shared Library:         yes                                                                                                                                                                                          
Static Library:         no                                                                                                                                                                                           
Extra libraries:                                                                                                                                                                                                     
                                                                                                                                                                                                                     
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/swapoff' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/swapoff'                                                                    
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/swapon' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/swapon'         
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/switch_root' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/switch_root' 
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/sync' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/sync'         
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/sysctl' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/sysctl'     
-- Configuring done                                                                                                                                                                                                  
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/syslogd' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/syslogd'
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tac' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tac'                
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tail' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tail'               
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tar' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tar'                                                                            
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/taskset' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/taskset'     
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tc' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tc'                                                                              
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tcpsvd' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tcpsvd'          
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/tee' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/tee'                     
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/telnet' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/telnet'                                                                      
-- Generating done                                                                                                                                                                                                   
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/telnetd' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/telnetd'          
'/nix/store/p93n9fwsrlrnp4szbz81s7i4ljl566y1-busybox-1.35.0/bin/test' -> '/nix/store/n94c6qkbfjpq4qkgh2f9xj5wv44m3154-extra-utils/bin/test'                                                                          
CMake Warning:                                                                                                                                                                                                       
  Manually-specified variables were not used by the project:                                                                                                                                                         
                                                                                                                                                                                                                     
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY               




(lots of output)


[79/80] Building CXX object cxx4/CMakeFiles/cxx4_test_att.dir/test_att.cpp.o                                                                                                                                                                                                                                                                                                                                                              
[80/80] Linking CXX executable cxx4/cxx4_test_att                                                                                                                                                                                                                                      
running tests                                                                                                                                                                                                                                                                                                                                                                                                                             
check flags: -j1 test                                                                                                                                                                                                                                                                                                                                                                                                                     
[0/1] Running tests...                                                                                                                                                                                                                                                                                                                                                                                                                    
Test project /build/source/build                                                                                                                                                                                                                                                                                                                                                                                                          
      Start  1: cxx4_test_classic                                                                                                                                                                                                                                                                                                                                                                                                         
 1/18 Test  #1: cxx4_test_classic ................   Passed    0.02 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  2: cxx4_test_group                                                                                                                                                                                                                                                                                                                                                                                                           
 2/18 Test  #2: cxx4_test_group ..................   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  3: cxx4_test_dim                                                                                                                                                                                                                                                                                                                                                                                                             
 3/18 Test  #3: cxx4_test_dim ....................   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  4: cxx4_test_att                                                                                                                                                                                                                                                                                                                                                                                                             
 4/18 Test  #4: cxx4_test_att ....................   Passed    0.02 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  5: cxx4_test_var                                                                                                                                                                                                                                                                                                                                                                                                             
 5/18 Test  #5: cxx4_test_var ....................   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  6: cxx4_test_var2                                                                                                                                                                                                                                                                                                                                                                                                            
 6/18 Test  #6: cxx4_test_var2 ...................   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  7: cxx4_test_type                                                                                                                                                                                                                                                                                                                                                                                                            
 7/18 Test  #7: cxx4_test_type ...................   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  8: cxx4_test_open_close                                                                                                                                                                                                                                                                                                                                                                                                      
 8/18 Test  #8: cxx4_test_open_close .............   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start  9: cxx4_test_ncFile_Flags                                                                                                                                                                                                                                                                                                                                                                                                    
 9/18 Test  #9: cxx4_test_ncFile_Flags ...........   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start 10: cxx4_test_filter                                                                                                                                                                                                                                                                                                                                                                                                          
10/18 Test #10: cxx4_test_filter .................***Failed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
Setting Filter....Caught unexpected exception.                                                                                                                                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                                                                                                                                          
      Start 11: examples_simple_xy_wr                                                                                                                                                                                                                                                                                                                                                                                                     
11/18 Test #11: examples_simple_xy_wr ............   Passed    0.01 sec                                                                                                                                                                                                                                                                                                                                                                   
      Start 12: examples_sfc_pres_temp_wr                                                                                                                                                                            
12/18 Test #12: examples_sfc_pres_temp_wr ........   Passed    0.01 sec                                                                                                                                              
      Start 13: examples_pres_temp_4D_wr                                                                                                                                                                                                                                                                                                                                                                                                  
13/18 Test #13: examples_pres_temp_4D_wr .........   Passed    0.01 sec                                                                                                                                              
      Start 14: examples_pres_temp_4D_rd                                                                                                                                                                             
14/18 Test #14: examples_pres_temp_4D_rd .........   Passed    0.01 sec                                                                                                                                              
      Start 15: examples_simple_xy_rd                                                                     
15/18 Test #15: examples_simple_xy_rd ............   Passed    0.01 sec                                                                                                                                              
      Start 16: examples_sfc_pres_temp_rd                                                                                                                                                                            
16/18 Test #16: examples_sfc_pres_temp_rd ........   Passed    0.01 sec                                                                                                                                              
      Start 17: examples_simple_xy_wr_formats                                                                                                                                                                        
17/18 Test #17: examples_simple_xy_wr_formats ....   Passed    0.01 sec                                                                                                                                              
      Start 18: examples_tst_filter                                                                                                                                                                                  
18/18 Test #18: examples_tst_filter ..............   Passed    0.03 sec                                                                                                                                              
                                                                                                                                                                                                                     
94% tests passed, 1 tests failed out of 18                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
Total Test time (real) =   0.21 sec                                                                                                                                                   
                                                                                                                                                                                                                     
The following tests FAILED:                                                                                                                                                                                                                                    
         10 - cxx4_test_filter (Failed)                                                                                                                                                                              
Errors while running CTest                                                                                                                                                                                                                                     
FAILED: CMakeFiles/test.util                                                                                                                                                                                         
cd /build/source/build && /nix/store/spkzmydsgmf0bm3xm4s8rqmmvvfaa42p-cmake-3.24.3/bin/ctest --force-new-ctest-process                                                                                                                                                                                                                                                       
ninja: build stopped: subcommand failed.                                                                                                                                                                                                                       
error: builder for '/nix/store/v5jb4wlnrciakfd4h6zi1hgb4rdfxwzx-netcdf-cxx4-4.3.1.drv' failed with exit code 1                                                                                                       
error: 1 dependencies of derivation '/nix/store/nyzh97zbkk1m1464k4xai6c86yd78mqd-nco-5.0.6.drv' failed to build                                                                                                                                                                                                                                                              
error: 1 dependencies of derivation '/nix/store/hz38nkjlkg4sb7g42w82ir49li7cblil-unit-sdc-nc-process.service.drv' failed to build                                                                                    
error: 1 dependencies of derivation '/nix/store/h2idjpby55kwyc9va0p4lbl8zc6dxvcy-system-units.drv' failed to build                                                                                                   
error: 1 dependencies of derivation '/nix/store/bsnqfvnm9n2jir6vkimgcm48bc2i5php-etc.drv' failed to build  
```


### Notify maintainers
@Stunkymonkey

### Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.63, NixOS, 22.05 (Quokka), 22.05.2809.7d7622909a3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.11-small"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/libraries/netcdf-cxx4/default.nix
+++ b/pkgs/development/libraries/netcdf-cxx4/default.nix
@@ -17,9 +17,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ cmake ninja ];
   buildInputs = [ netcdf hdf5 curl ];
 
-  # 10 - cxx4_test_filter (Failed)
-  # Setting Filter....Caught unexpected exception.
-  doCheck = false;
+  doCheck = true;
   enableParallelChecking = false;
   preCheck = ''
     export HDF5_PLUGIN_PATH=${netcdf}/lib/hdf5-plugins

--- a/pkgs/development/libraries/netcdf-cxx4/default.nix
+++ b/pkgs/development/libraries/netcdf-cxx4/default.nix
@@ -21,6 +21,9 @@ stdenv.mkDerivation rec {
   # Setting Filter....Caught unexpected exception.
   doCheck = false;
   enableParallelChecking = false;
+  preCheck = ''
+    export HDF5_PLUGIN_PATH=${netcdf}/lib/hdf5-plugins
+  '';
 
   meta = {
     description = "C++ API to manipulate netcdf files";

--- a/pkgs/development/libraries/netcdf/default.nix
+++ b/pkgs/development/libraries/netcdf/default.nix
@@ -56,6 +56,8 @@ in stdenv.mkDerivation rec {
   ]
   ++ (lib.optionals mpiSupport [ "--enable-parallel-tests" "CC=${mpi}/bin/mpicc" ]);
 
+  enableParallelBuilding = true;
+
   disallowedReferences = [ stdenv.cc ];
 
   postFixup = ''

--- a/pkgs/development/libraries/netcdf/default.nix
+++ b/pkgs/development/libraries/netcdf/default.nix
@@ -53,6 +53,7 @@ in stdenv.mkDerivation rec {
       "--enable-dap"
       "--enable-shared"
       "--disable-dap-remote-tests"
+      "--with-plugin-dir=${placeholder "out"}/lib/hdf5-plugins"
   ]
   ++ (lib.optionals mpiSupport [ "--enable-parallel-tests" "CC=${mpi}/bin/mpicc" ]);
 

--- a/pkgs/development/libraries/netcdf/default.nix
+++ b/pkgs/development/libraries/netcdf/default.nix
@@ -1,6 +1,11 @@
 { lib, stdenv
 , fetchurl, unzip
 , hdf5
+, bzip2
+, libzip
+, zstd
+, szipSupport ? false
+, szip
 , libxml2
 , m4
 , curl # for DAP
@@ -34,7 +39,10 @@ in stdenv.mkDerivation rec {
     hdf5
     libxml2
     mpi
-  ];
+    bzip2
+    libzip
+    zstd
+  ] ++ lib.optional szipSupport szip;
 
   passthru = {
     inherit mpiSupport mpi;

